### ISSUE TITLE
useState with [] causes multiple subscriptions

### ISSUE THREAD

**Issue by cebaa** [MatthiasKainer/pure-lit#27]
**Describe the bug**
Calling `subscribe` on `useState(element, [])` causes multiple subscriptions

**To Reproduce**
```
  pureLit("demo-element", 
    element => {
      const {set, subscribe} = useState(element, [])
      console.log('demo-element')
      subscribe(state => console.log('new state', state))
      return html`<button @click=${() => set(["lala"])}>Add value</button>`
     }
   )
```

after three clicks logs this:

```
demo-element
new state ['lala']
demo-element
new state ['lala']
new state ['lala']
demo-element
new state ['lala']
new state ['lala']
new state ['lala']
demo-element
```

**Expected behavior**
It should subscribe only once.

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: 127

**Additional context**
The issue does not appear with `useState(element, 0)`, i.e. int instead of an array.

**Comment by MatthiasKainer** [comment]
Opened the bug at https://github.com/MatthiasKainer/lit-element-state-decoupler which is the underlying lib, and will try to fix it there

**Comment by MatthiasKainer** [comment]
The change fixed it, but the newest version of lit-element-state-decoupler is not yet available in the pipeline due to eventual consistency on npm. I'll retrigger later, then it will be fixed with v3.0.5

**Comment by MatthiasKainer** [comment]
pushed, if an issue pops up with v3.0.5 feel free to create another issue

thx
